### PR TITLE
Fix export snapshot

### PR DESCRIPTION
**Pull request by depahelix2021**
## Types of changes

- [] Bug fix (non-breaking change which fixes an issue)
- [] Improvement (non-breaking change which improves existing functionality)
- [] New feature (non-breaking change which adds new functionality)
- [] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [] My code follows the code style of this project.
- [] My change requires a change to the documentation.
- [] I have updated the documentation accordingly.
- [] I have read the **CONTRIBUTING** document.
- [] I have added tests to cover my changes.
- [] All new and existing tests passed.

### CODE REVIEW

File: app/assets/javascripts/services/caseCSVSvc.js [hunk 1]
```diff
@@ -351,35 +377,41 @@
          */
 
         function stringifySnapshot (aCase, snapshot, withHeader) {
-          var csvContent  = '';
-
+          let infoArray = [];
```
Comment (epugh) [inline app/assets/javascripts/services/caseCSVSvc.js:380]: nice to get these JS things fixed!

File: app/assets/javascripts/services/caseCSVSvc.js [hunk 2]
```diff
@@ -351,35 +377,41 @@
...
+          let matchingQuery = null;
+          let matchingQueryText = null;
+          const snapshotName = snapshot.name();
+          const snapshotTime = snapshot.time;
+          const caseNumber = aCase.caseNo;
+          let csvContent = '';
+          let queryIdAsInt = 0;
```
Comment (epugh) [inline app/assets/javascripts/services/caseCSVSvc.js:387]: do we actually have to provide a default like this?   Versus defining it where we use it?

File: app/assets/javascripts/services/caseCSVSvc.js [hunk 3]
```diff
@@ -351,35 +377,41 @@
...
-              var query = snapshot.queries.filter(function(q) { return q.queryId === queryId; })[0];
-
-              infoArray.push(stringifyField(query.query_text));
-              infoArray.push(stringifyField(doc.id));
-              infoArray.push(stringifyField(idx+1));
-
-              dataString = infoArray.join(',');
-              csvContent += dataString + EOL;
+          angular.forEach(snapshot.docs, function (docs, queryId) {
+            queryIdAsInt = parseInt(queryId, 10);
```
Comment (epugh) [inline app/assets/javascripts/services/caseCSVSvc.js:394]: i think this should be where the "let" happens

Comment (depahelix2021) [reply app/assets/javascripts/services/caseCSVSvc.js]: queryId is not known until this line, comes in from L393.  If you mean that you want to add a let at the beginning of L394 and take out L387, that would work.

File: app/assets/javascripts/services/snapshotFactory.js [hunk 4]
```diff
@@ -26,6 +26,12 @@
       self.timestamp        = timestamp;
 
       self.docIdsPerQuery   = {};
+      
+      // Map from snake_case to camelCase.
+      angular.forEach(self.queries, function(query) {
+        query.queryId = query.query_id;
+        delete query.query_id;
+      });
```
[the remark this answers is not in the document]
Comment (epugh) [reply app/assets/javascripts/services/snapshotFactory.js:34]: Nope!  This is an established pattern....  (you just didn't know about it!).

[the remark this answers is not in the document]
Comment (depahelix2021) [reply app/assets/javascripts/services/snapshotFactory.js:34]: OK

**Review by depahelix2021 — COMMENTED** [review]
LGTM

**Pull request merged**